src/Interfaces/RequestSender.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AshAllenDesign\LaravelExchangeRates\Interfaces;
+
+use Illuminate\Http\Client\RequestException;
+
+interface RequestSender
+{
+    /**
```

Comment:
Maybe it's a good moment to include a ResponseParser in order to have an abstraction layer between `RequestSender` and the `SharedDriverLogicHandler`.

In the `SharedDriverLogicHandler` line 125 for example, the class is reading directly from the response a 'rates' keys assuming that all drivers `RequestSender`'s implementations will return that but in the code there's no indication of what expects. 

Proposal:
```php
interface ResponseParser {

public function rates(): array;

public function raw(): mixed;

}
```

Reply:
This is something I hadn't even considered but definitely think would be worthwhile! Is this something you'd be open to adding yourself via a PR? Of course, if you'd prefer not to, that's totally cool and I can get it added 🙂

Reply:
Yes, sure. Will take a look asap 💪🏽

Reply:
Thinking about it, I think I might hold off on this particular feature. I do think it could have a place in future versions of the package, but for the time being, I don't think it's necessarily needed because quite a few of the APIs I'm intending on supporting follow the same response structure.

But I definitely agree this is something that could be useful in the future for supporting APIs that have a different structure.

So if you'd like to PR this functionality further down the line, it's definitely something that I'll be open to reviewing 😄

Reply:
Sounds good! 💯